### ISSUE TITLE
(Mixed mode) clicking on a file should open its details in the next column

### ISSUE THREAD

**Issue by hanaral** [elementary/files#1601]
## Problem

When in column view, clicking a folder lists its contents in the next column over. However, doing the same to a file just opens the file rather than listing a preview said files contents. 

## Proposal

The next column over should display the preview of the file when it is clicked, optimally with images and videos using the same fast preview image generation as the current previews used in icons. 
I know this idea has been tossed around and that there is a branch for a sidebar, but that defeats the purpose almost entirely. 
The point of the macOS column view is to list or preview an items contents for both files and directories in the same way, not to have one of them use the column system and the other to have a permenant section of the UI filled up with something better of as a seperate window.

## Prior Art

![Screenshot from 2021-02-25 14-51-54](https://user-images.githubusercontent.com/58219504/109171288-891c1800-7779-11eb-976e-122082d34f08.png)
_This also illustrates the difference between what a breif preview means and what a properties menu looks like_

**Comment by cassidyjames** [comment]
Yes, with mixed mode it makes sense to put file properties into a column/right sidebar somehow as mentioned [here](https://github.com/elementary/files/pull/1036#issuecomment-717460509):

>I have been trying this for a bit and really like where it's going UX-wise. I think it will feel even better if we get a Properties sidebar on file select, as then something still happens when clicking files. Of course something to figure out there will be if the sidebar causes a reflow, or overlays, or what. But I do think that can be addressed in a future branch.

We still need to sort out how this works in the various views, i.e. if it overlays or reflows or expands the window, or what.

**Comment by hanaral** [comment]
I should probably reiterate that it makes more logical sense and takes up significantly less UI room to add the preview of the file directly in to the next column, even if that sort of implementation would require more effort than just adding a sidebar.

**Comment by jeremypw** [comment]
The view widget (Slot) does have the functionality to show extra widgets in addition to or instead of the directory view (AbstractDirectoryView), at the moment used for e.g. Trash action bar. Show showing a properties widget in a column in ColumnView should therefore not be a problem.  The other views would have to be modified somewhat to be able to show an extra widget to the side rather than at the top or bottom but it should be possible.

**Comment by jeremypw** [comment]
Need to get a good design done before leaping into coding though, of course.

**Comment by shiruken1** [comment]
I volunteer as a tribute ✋ 

I've been wanting this feature bad enough for a _while_ now and figured I'd find a ticket around here somewhere asking for this, so I'mma go learn Vala...

**Comment by jeremypw** [comment]
@shiruken1 Thanks - we love a volunteer!  @danirabbit Are you happy for this feature to be implemented?
